src/sql/workbench/contrib/jobManagement/browser/jobStepsView.component.ts

```diff
@@ -105,6 +107,10 @@ export class JobStepsViewComponent extends JobManagementView implements OnInit,
 		this._register(attachListStyler(this._tree, this.themeService));
 		const stepsTooltip = nls.localize('agent.steps', "Steps");
 		jQuery('.steps-header > .steps-icon').attr('title', stepsTooltip);
+		this._jobManagementService.stepsChanged((data: JobStepsViewRow[]) => {
```

Comment:
Why go through the service for this? Seems like you could just have the emitter be on the jobHistory component instead and take out the middle man since nothing else needs to use this.

Reply:
The steps tree is a grandchild and you need a service to communicate between parents and grandchildren. We can't get rid of the middle element because we use it as a container for the tree.

Reply:
What do you mean - you already have a reference to the jobHistoryComponent (see line 100). If that had an event emitter then you could just use that. 

Anyways I'm no expert on angular but I don't really think we're doing it "correctly" here anyways. Having a more model view approach where there's a single model containing the data that is updated and then having the UI components update on that seems like it would be the more correct way. But that's obviously outside the scope here - I just don't think there's any solid pattern that we follow with this stuff currently as it is.

Reply:
@Charles-Gagnon I agree with what you're saying but if we take the event directly from jobHistoryComponent and use it and do the same in future, that would make the code a lot more complicated for data passing between components. A single service for data passing between components is ideal. Infact, in a future change I should remove the reference to jobHistoryComponent and use the service instead.